### PR TITLE
Use new client-go / k8s.io/api

### PR DESCRIPTION
**Pull request by bcreane**
Use new client-go / k8s.io/api

## Todos
- [X] Tests

## Release Note

```release-note
None required
```

### CODE REVIEW

File: lib/apis/v2/register.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+// Resource takes an unqualified resource and returns a Group qualified GroupResource
+func Resource(resource string) schema.GroupResource {
+	return SchemeGroupVersion.WithResource(resource).GroupResource()
+}
+
+// Adds the list of known types to api.Scheme.
+func addKnownTypes(scheme *runtime.Scheme) error {
+	scheme.AddKnownTypes(SchemeGroupVersion,
+		&BGPConfiguration{},
```
Comment (caseydavenport) [inline lib/apis/v2/register.go]: @robbrockbank @bcreane do we need these here at all? Do we need more here?

Comment (bcreane) [reply lib/apis/v2/register.go]: looks like we don't need either BGPConfiguration{}, or BGPConfigurationList{}.
I can clear more out of register.go tomorrow, if that works for you?

Comment (bcreane) [reply lib/apis/v2/register.go]: note that we do need most of register.go for linking to succeed (I'll push an update tomorrow that elides BGPConfiguration structs).

File: lib/apis/v2/bgpconfig.go [hunk 2]
```diff
@@ -25,6 +25,8 @@ const (
 	KindBGPConfigurationList = "BGPConfigurationList"
 )
 
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
```
Comment (caseydavenport) [inline lib/apis/v2/bgpconfig.go]: I think we should have newlines between the generator tags and our other comments, throughout this PR.

Comment (bcreane) [reply lib/apis/v2/bgpconfig.go]: yep, agreed. putting together a PR right now.

Comment (bcreane) [reply lib/apis/v2/bgpconfig.go]: note that I squashed the commits, its newlines + stripped down register.go

File: lib/apis/v2/register.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+// Resource takes an unqualified resource and returns a Group qualified GroupResource
+func Resource(resource string) schema.GroupResource {
+	return SchemeGroupVersion.WithResource(resource).GroupResource()
+}
+
+// Adds the list of known types to api.Scheme.
+func addKnownTypes(scheme *runtime.Scheme) error {
+	scheme.AddKnownTypes(SchemeGroupVersion,
```
Comment (bcreane) [inline lib/apis/v2/register.go]: ugh, this should probably just return `nil`

**Comment by bcreane** [comment]
Thanks for merging Shaun.

**Pull request merged**